src/Orc.CsvTextEditor/CsvTextEditorTools/CsvTextEditorToolBase.cs

```diff
@@ -30,21 +23,17 @@ public CsvTextEditorToolBase()
         #endregion
 
         #region Methods
-        public bool IsInitialized(object scope = null)
-        {
-            return _initializedInScopes.Contains(scope);
-        }
 
         public abstract void Open();
         public abstract void Close();
 
-        public void Initialize(TextEditor textEditor, object scope)
+        public void Initialize(TextEditor textEditor, ICsvTextEditorService csvTextEditorService)
```

Comment:
I think injecting service here is not good idea
I think we must have one instance of ICsvTextEditorTool per ICsvTextEditorService. and inject the service into ctor
don't you agree?